File: 0933-number-of-recent-calls/0933-number-of-recent-calls.py
class RecentCounter:

    def __init__(self):
        self.c =0
        self.q = []

    def ping(self, t: int) -> int:
        self.q.append(t)
        v=0
        while (True):
            v = self.q[0]
            if v>=(t-3000):
                break
            else:
                self.q.pop(0)
        return len(self.q)

# Your RecentCounter object will be instantiated and called as such:
# obj = RecentCounter()
# param_1 = obj.ping(t)

File: 0933-number-of-recent-calls/0933-number-of-recent-calls.java
class RecentCounter {
    Queue<Integer> q;
    int c;
    public RecentCounter() {
        this.c=0;
        this.q = new LinkedList<>();
    }
    
    public int ping(int t) {
        this.q.offer(t);
        while(!this.q.isEmpty()){
            int v = this.q.peek();
            if (v>= t-3000){
                break;
            }else{
                this.q.poll();
            }
        }
        return this.q.size();
    }
}

/**
 * Your RecentCounter object will be instantiated and called as such:
 * RecentCounter obj = new RecentCounter();
 * int param_1 = obj.ping(t);
 */